vpc/bpf2/filter.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Compilation instructions:
+ * clang -O2 -emit-llvm -c filter.c -o - | llc -march=bpf -filetype=obj -o filter.o
+ */
+#include <stdint.h>
+#include <asm/types.h>
+
+#include <linux/bpf.h>
+#include <linux/pkt_sched.h>
+#include <linux/if_ether.h>
+#include <linux/ip.h>
+#include <linux/ipv6.h>
+#include <linux/pkt_cls.h>
+#include <string.h>
+#include "bpf_api.h"
+
+struct ipv4_key {
+	__u32		vlan_tci;
+	__be32		addr;
+};
+
+struct bpf_elf_map __section_maps ipv4_map = {
+        .type           = BPF_MAP_TYPE_HASH,
+        .size_key       = sizeof(struct ipv4_key),
+        .size_value     = sizeof(__u16),
+        .pinning        = PIN_GLOBAL_NS,
+        .max_elem       = 10240,
```

Comment:
Why 10240?

Reply:
It leaves enough room for leaks, that we wouldn't notice for a while.